File: commons/tool/src/webapp/js/commons_permissions.js
function CommonsPermissions(data) {

	if (!data) return;

	this.postReadAny = false;
	this.postCreate = false;
	this.postDeleteAny = false;
	this.postDeleteOwn = false;
	this.postUpdateAny = false;
	this.postUpdateOwn = false;
	this.commentCreate = false;
	this.commentDeleteAny = false;
	this.commentDeleteOwn = false;
	this.commentUpdateAny = false;
	this.commentUpdateOwn = false;
	this.updateSite = false;

	for(var i=0,j=data.length;i<j;i++) {
		if('commons.post.read.any' === data[i])
			this.postReadAny = true;
		else if('commons.post.create' === data[i])
			this.postCreate = true;
		else if('commons.post.delete.any' === data[i])
			this.postDeleteAny = true;
		else if('commons.post.delete.own' === data[i])
			this.postDeleteOwn = true;
		else if('commons.post.update.any' === data[i])
			this.postUpdateAny = true;
		else if('commons.post.update.own' === data[i])
			this.postUpdateOwn = true;
		else if('commons.comment.create' === data[i])
			this.commentCreate = true;
		else if('commons.comment.delete.any' === data[i])
			this.commentDeleteAny = true;
		else if('commons.comment.delete.own' === data[i])
			this.commentDeleteOwn = true;
		else if('commons.comment.update.any' === data[i])
			this.commentUpdateAny = true;
		else if('commons.comment.update.own' === data[i])
			this.commentUpdateOwn = true;
		else if('site.upd' === data[i])
			this.updateSite = true;
	}
}

File: commons/api/src/java/org/sakaiproject/commons/api/datamodel/CommonsPermissions.java
/**
 * Copyright (c) 2003-2017 The Apereo Foundation
 *
 * Licensed under the Educational Community License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *             http://opensource.org/licenses/ecl2
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.sakaiproject.commons.api.datamodel;

import lombok.Getter;
import lombok.Setter;

@Getter @Setter
public class CommonsPermissions {

    private String role;
    private boolean postCreate = false;
    private boolean postReadAny = false;
    private boolean postReadOwn = false;
    private boolean postUpdateAny = false;
    private boolean postUpdateOwn = false;
    private boolean postDeleteAny = false;
    private boolean postDeleteOwn = false;
    private boolean commentCreate = false;
    private boolean commentReadAny = false;
    private boolean commentReadOwn = false;
    private boolean commentUpdateAny = false;
    private boolean commentUpdateOwn = false;
    private boolean commentDeleteAny = false;
    private boolean commentDeleteOwn = false;
}
